### PR TITLE
[Issue-1002] Add script for issue 1002 [internal review]

### PR DESCRIPTION
**Pull request by VjKlepikov**
ATF Test Scripts to check [FORDTCN-11982](https://adc.luxoft.com/jira/browse/FORDTCN-11982)

This PR is **[ready]** for review.

### Summary
New ATF scripts to check the issue.

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    and data.serviceType == pServiceId
+    and data.sessionId == common.getMobileSession().mobile_session_impl.control_services.session.sessionId.get()
+    and data.frameInfo == constants.FRAME_INFO.END_SERVICE
+  end
+  return common.getMobileSession():ExpectEvent(event, "EndService")
+end
+
+local function endServiceByUserExit()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnExitApplication",
+  { appID = common.app.getHMIId(), reason = "USER_EXIT" })
```
Comment (dboltovskyi) [inline test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua:77]: For better stability I would move this action to the end of function (before any expectation)

File: test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+-- 2) Navigation App is registered and activated
+-- 3) Video streaming is started
+-- 4) Audio streaming is started
+-- In case:
+-- 1) User Exit from App
+-- SDL does:
+-- send Navigation.OnAudioDataStreaming(available = false) notification to HMI
+-- send Navigation.OnVideoDataStreaming(available = false) notification to HMI
+-- send Navigation.StopAudioStream request to HMI
+-- send Navigation.StopStream request to HMI
```
Comment (dboltovskyi) [inline test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua:18]: Please add list dashes. E.g.:
```
SDL does:
 - expectation 1
 - expectation 2
```

File: test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  common.getHMIConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = false })
+  :Times(AtLeast(1))
+  common.getHMIConnection():ExpectRequest("Navigation.StopAudioStream", { appID = common.getHMIAppId() })
+  :Do(function(_, data)
+    common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  common.getHMIConnection():ExpectRequest("Navigation.StopStream", { appID = common.getHMIAppId() })
+  :Do(function(_, data)
+    common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
```
Comment (dboltovskyi) [inline test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua:89]: I would correct indentations across all script

File: test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  return common.getMobileSession():ExpectEvent(event, "EndService")
+end
+
+local function endServiceByUserExit()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnExitApplication",
+  { appID = common.app.getHMIId(), reason = "USER_EXIT" })
+  common.getHMIConnection():ExpectNotification("Navigation.OnAudioDataStreaming", { available = false })
+  :Times(AtLeast(1))
+  common.getHMIConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = false })
+  :Times(AtLeast(1))
```
Comment (dboltovskyi) [inline test_scripts/Defects/5_1/1002_SDL_sends_StopAudioStream_in_case_User_Exit_during_Audio_Video_streaming.lua:81]: Not sure whether we have to use `AtLeast` here

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

**Comment by VjKlepikov** [comment]
@dboltovskyi Thank you for the review, all comments have been processed

**Review by dboltovskyi — APPROVED** [review]
@VjKlepikov Thank you for updates

**Pull request merged**